apps/web/components/Embed.tsx

```diff
@@ -844,6 +1164,23 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
     },
   });
 
+  const handleCopyEmailText = () => {
+    const contentElement = emailContentRef.current;
+    if (contentElement !== null) {
+      const range = document.createRange();
+      range.selectNode(contentElement);
+      const selection = window.getSelection();
+      if (selection) {
+        selection.removeAllRanges();
+        selection.addRange(range);
+        document.execCommand("copy");
+        selection.removeAllRanges();
+      }
+
+      showToast(t("code_copied"), "success");
+    }
+  };
```

Comment:
This function copies all the contents from our email preview element using ref and range